src/lib/resource.js

```diff
@@ -1,51 +1,121 @@
-    }
-    logger.info(`Resource Queue: Successfully retrieved resource metadata for resource '${resourceUrl}'`);
-    return ({ status: httpCodes.OK, data: { url: resourceUrl } });
-  } catch (error) {
-    auditLogger.error(`Resource Queue: Unable to retrieve title for Resource (ID: ${resourceId} URL: ${resourceUrl}), please make sure this is a valid address:`, error);
-    throw Error(error); // We must rethrow the error here to ensure the job is retried.
-  }
-};
-
-export {
-  axios,
-  getResourceMetaDataJob,
-};
+/* eslint-disable import/prefer-default-export */
+import httpCodes from 'http-codes';
+import axios from 'axios';
+import { auditLogger, logger } from '../logger';
+import { Resource } from '../models';
+
+const getPageScrapeValues = async (resourceUrl, isEclkc) => {
+  // Use axios to get url info.
+  const res = await axios.get(resourceUrl, { maxRedirects: 3 });
+
+  // Scrape Title & National Centers from ECLKC resources.
+  const foundTitle = res.data.match(/<title[^>]*>([^<]+)<\/title>/);
+
+  // If we found a title update it.
+  if (foundTitle && foundTitle.length >= 1) {
+    // Get title.
+    const titleToUpdate = foundTitle[1].trim();
+
+    // Update db.
+    const updatedCnt = await Resource.update({
+      title: titleToUpdate,
+    }, {
+      where: { url: resourceUrl },
+      individualHooks: false,
+    });
+
+    // If we don't update anything throw an error to retry.
+    if (updatedCnt[0] === 0) {
+      throw Error('Failed to update db resource.');
+    }
+
+    return updatedCnt[0];
+  }
+
+  return 0;
+};
+
+const getMetadataValues = async (resourceId, resourceUrl) => {
+  // Attempt to get the resource metadata (if valid ECLKC resource).
+  // Sample: https://eclkc.ohs.acf.hhs.gov/mental-health/article/head-start-heals-campaign?_format=json
+  try {
+    const metadataUrl = `${resourceUrl}?_format=json`;
+    const res = await axios.get(metadataUrl, { maxRedirects: 3 });
+    const metadata = res.data;
+
+    // get created.
+    const { created } = metadata;
+    // get changed.
+    const { changed } = metadata;
+    // get title.
+    const { title } = metadata;
+    // get field_taxonomy_national_centers.
+    // eslint-disable-next-line max-len
+    const fieldTaxonomyNationalCenters = metadata.field_taxonomy_national_centers;
+    // get the field_taxonomy_topic.
+    const fieldTaxonomyTopic = metadata.field_taxonomy_topic;
+    // get the langcode.
+    const { langcode } = metadata;
+    // get the field_context.
+    const fieldContext = metadata.field_context;
+
+    // Create the metadata object.
+    const metadataObj = {
+      created,
+      changed,
+      title,
+      fieldTaxonomyNationalCenters,
+      fieldTaxonomyTopic,
+      langcode,
+      fieldContext,
+    };
+
+    // Update URL in DB.
+    await Resource.update({
+      metadata: metadataObj,
+      metadataUpdatedAt: new Date(),
+    }, {
+      where: { url: resourceUrl },
+      individualHooks: false,
+    });

```

Comment:
are we okay with any/all of these being potentially undefined? do they need fallback values before the update happens?

most of these property names are self-explanatory, but I'm not sure I know what `langcode` and `fieldContext` are, so maybe the `get langcode.` comment could be expanded to something like `get langcode - <description of what langcode is and why its important that we store it>`

Reply:
I don't have a default definition for each field. IE I would assume 'langcode' is language code but how its used I'm not 100% certain. I would think we take the values as is do you seeing it causing issue? We are still figuring out how all this will be used so I would imagine we would check for undefined.

Reply:
ah okay, so we're just sortof grabbing everything and figuring out later if we need it or not

in that case, is there a reason we don't just grab everything from `metadata`? e.g.

```javascript
await Resource.update({
  metadata: { ...metadata },
}, { ... });
```

or is it large enough that we don't want to store the entire thing?